diff -rupN memleax-1.1.1-orig/configure memleax-1.1.1/configure
--- memleax-1.1.1-orig/configure	2018-01-04 06:05:08.000000000 +0800
+++ memleax-1.1.1/configure	2022-03-21 11:01:15.287819520 +0800
@@ -2,6 +2,8 @@
 
 disable_dwarf=false
 prefix='/usr'
+LLFLAG=""
+CFLAGS=""
 
 # parse options
 for opt in $@; do
@@ -14,6 +16,12 @@ for opt in $@; do
 	'--prefix')
 		prefix=$value
 		;;
+	'--LLFLAG')
+		LLFLAG="$LLFLAG $value"
+		;;
+	'--CFLAGS')
+		CFLAGS="$CFLAGS $value"
+		;;
 	'--version')
 		echo "version: 1.1.1"
 		exit 0
@@ -26,6 +34,8 @@ for opt in $@; do
 		echo "        and you will not see file name and line number in backtrace."
 		echo "  --prefix=<path>"
 		echo "        Set install path. default is '/usr/'."
+		echo "  --LLFLAG=<path>"
+		echo "  --CFLAGS=<path>"
 		echo "  --version"
 		echo "        Show version."
 		echo "  --help"
@@ -43,14 +53,6 @@ done
 # check OS and machine type
 echo 'checking machine type...'
 case `uname -m` in
-'x86'|'i386'|'i686')
-	machine="x86"
-	CFLAGS="$CFLAGS -DMLX_X86"
-	;;
-'x86_64'|'amd64')
-	machine="x86_64"
-	CFLAGS="$CFLAGS -DMLX_X86_64"
-	;;
 'armv7'*)
 	machine="arm"
 	CFLAGS="$CFLAGS -DMLX_ARMv7"
@@ -60,11 +62,13 @@ case `uname -m` in
 	CFLAGS="$CFLAGS -DMLX_AARCH64"
 	;;
 *)
-	echo "Error: unsupported machine: `uname -m`."
+	#echo "Error: unsupported machine: `uname -m`."
 	echo "Only x86, x86_64, arm, and aarch64 are supported."
-	exit 2
+	#exit 2
 	;;
 esac
+machine="arm"
+CFLAGS="$CFLAGS -DMLX_ARMv7"
 
 echo 'checking OS type...'
 case `uname` in
@@ -84,70 +88,8 @@ case `uname` in
 	;;
 esac
 
-
-# check libraries
-echo 'checking libraries...'
-check_lib()
-{
-	lib=$1
-	header=$2
-	function=$3
-	cflags=$4
-	ldflags=$5
-
-	echo "checking $lib..."
-	cat << EOF > tmp.c
-		#include <$header>
-		#include <stdio.h>
-		int main() {
-			$function;
-			return 0;
-		}
-EOF
-	if ! cc -M $CFLAGS $cflags tmp.c >/dev/null 2>&1 ; then
-		echo "$lib-devel is missing."
-		rm -f tmp.c
-		return 1
-	fi
-	if ! cc $CFLAGS $cflags tmp.c -o tmp $LDFLAGS $LDLIBS $ldflags >/dev/null 2>&1 ; then
-		echo "$lib is missing."
-		rm -f tmp.c tmp
-		return 1
-	fi
-	rm -f tmp.c tmp
-	return 0
-}
-
-# - check libunwind
-LDLIBS="$LDLIBS -lunwind-ptrace -lunwind -lunwind-$machine"
-if ! check_lib libunwind libunwind.h "unw_create_addr_space(NULL,0)" ; then
-	echo "Error: libunwind-devel is required."
-	exit 1
-fi
-
 # - check libelf
-LDLIBS="$LDLIBS -lelf"
-if ! check_lib libelf libelf.h "elf_begin(0,0,NULL)" ; then
-	echo "Error: libelf-devel is required."
-	exit 1
-fi
-
-# - check libdwarf
-if ! $disable_dwarf; then
-	if check_lib libdw libdw.h "dwarf_begin(0,0)" "-I/usr/include/elfutils" "-ldw"; then
-		CFLAGS="$CFLAGS -DMLX_WITH_LIBDW -I/usr/include/elfutils/"
-		LDLIBS="$LDLIBS -ldw"
-	elif check_lib libdwarf libdwarf.h "dwarf_init(0,0,0,0,NULL,NULL)" "-I/usr/include/libdwarf" "-ldwarf"; then
-		CFLAGS="$CFLAGS -DMLX_WITH_LIBDWARF -I/usr/include/libdwarf/"
-		LDLIBS="$LDLIBS -ldwarf"
-	else
-		echo "Error: libdw or libdwarf is required."
-		echo "You could set '--disable-debug_line' to disable it. As a result,"
-		echo "the file name and line number will not be showed in backtrace."
-		exit 1
-	fi
-fi
-
+LDLIBS="$LDLIBS $LLFLAG -lelf -lunwind-ptrace -lunwind -lunwind-$machine -ldw"
 
 # check done, generate Makefile
 echo 'generate Makefile...'
@@ -159,7 +101,7 @@ LDLIBS = $LDLIBS
 LDFLAGS = $LDFLAGS
 PREFIX = $prefix
 DESTDIR =
-
+CC="/opt/toolchains//crosstools-arm-gcc-5.5-linux-4.1-glibc-2.26-binutils-2.28.1/usr/bin/arm-buildroot-linux-gnueabi-gcc"
 TARGET = memleax
 
 SOURCES = $SOURCES
@@ -169,7 +111,7 @@ EOF
 
 cat << 'EOF' >> Makefile
 $(TARGET) : $(OBJS)
-	$(CC) -o $@ $(OBJS) $(LDFLAGS) $(LDLIBS)
+	$(CC) -o $@ $(OBJS) $(CFLAGS) $(LDFLAGS) $(LDLIBS)
 
 clean :
 	rm -f $(OBJS) $(TARGET)
